### ISSUE TITLE
call to consumer.committed may segfault (macOS)

### ISSUE THREAD

**Issue by edoardocomar** [Blizzard/node-rdkafka#285]
Running a consumer which in its `on('data',..)` calls `committed(...)`
may segfault (on macOS) if the `committed` function is invoked many times in rapid succession.

We are attaching a somewhat artificial example where committed is invoked in a loop, 
which may be easier to reproduce, by just sending one message to the topic

but the segfault happens as well in real use cases where committed is invoked just once in `on('data'...)` but the consumer receives a lot of messages in quick succession.

investigated with @mimaison

**Comment by edoardocomar** [comment]
```
var Kafka = require('../');

var consumer = new Kafka.KafkaConsumer({
  'metadata.broker.list': 'localhost:9092',
  'group.id': 'consumerMH.group.id' + Date.now().toString(),
  'enable.auto.commit': false,
});

var topicName = 'testtopic';

consumer.on('ready', function(arg) {
  console.log('consumer ready.' + JSON.stringify(arg));
  consumer.subscribe([ topicName ]);
  consumer.consume();
});

consumer.on('data', function(m) {
   console.log("received:", JSON.stringify(m));
 
   for(var j=0; j<1000; j++) {
    consumer.committed(5000, function(err, topicParts) {
      console.log("committed cb: "+ JSON.stringify(err) + " " + JSON.stringify(topicParts))
    });
  }
});

consumer.connect();
```

**Comment by webmakersteve** [comment]
I believe I fixed it with this: 7423e540baffb685062c887c70a792ab6ad4a34f

I changed the implementation so you need to provide the topic partitions you want to get committed offsets or position for. If you provide something falsy, it defaults to the current assignment set, but you need to explicitly null the first parameter to get that behavior.

**Comment by webmakersteve** [comment]
Closing because I believe it is fixed and the issue has been inactive.
